src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -210,19 +214,53 @@ public boolean isValidIdentifierForTerm(SQLIdentifierExpr expr) {
             );
     }
 
-    public boolean hasIdenticalMappings(TermFieldScope scope, Map<String, String> indexToType) {
+    public void checkMappingCompatibility(TermFieldScope scope, Map<String, String> indexToType) {
         if (scope.getMapper().isEmpty()) {
             throw new VerificationException("Unknown index " + indexToType.keySet());
         }
+        // Collect all FieldMappings into hash set and ignore index/type names. set.size > 1 means FieldMappings NOT unique,
+        // needs further compatibility check
+        Set<FieldMappings> set = curScope().getMapper().allMappings().stream().
+                                 flatMap(typeMappings -> typeMappings.allMappings().stream()).
+                                 collect(Collectors.toSet());
 
-        if (isMappingOfAllIndicesDifferent()) {
-            return false;
-        }
+        FieldMappings fieldMappings = null;
+
+        if (set.size() > 1) {
+            Map<String, Map<String, Object>> map = new LinkedHashMap<>();
+
+            for (FieldMappings f : set) {
+                Map<String, Map<String, Object>> m = f.data();
+
+                for (Map.Entry<String, Map<String, Object>> e : m.entrySet()) {
+                    String fieldName = e.getKey();
+                    Map<String, Object> fieldMappingValue = e.getValue();
+
+                    if (!map.containsKey(fieldName)) {
+                        map.put(fieldName, fieldMappingValue);
+                    } else {
+                        // check if types are same
+                        if (!fieldMappingValue.equals(map.get(fieldName))) {
+                           // TODO: Merge mappings if they are compatible, like text and text/keyword to text/keyword.
 
+                            String firstFieldType= new JSONObject(fieldMappingValue).toString().replaceAll("\"", "");
+                            String secondFieldType = new JSONObject(map.get(fieldName)).toString().replaceAll("\"", "");
+
+                            String exceptionReason = String.format("Cannot use field [%s] " +
+                                            "due to ambiguities being mapped as incompatible types: [%s] and [%s] ",
```

Comment:
not sure what 'ambiguities' refers to here, but I think the message itself is ambiguous. +Ideally we provide the index name also (but that's not a blocker). Maybe just say `"Field [%s] has different different mappings in the specified indices. Index [%s] - mapping: [%s], index [%s] - mapping [%s]"`. If you want to omit index names, maybe something like `"Different mappings are not allowed for the same field: found [%s] and [%s]"`. Finally, please use `Locale.ROOT` with `String.format`.

Reply:
> not sure what 'ambiguities' refers to here, but I think the message itself is ambiguous. +Ideally we provide the index name also (but that's not a blocker). Maybe just say `"Field [%s] has different different mappings in the specified indices. Index [%s] - mapping: [%s], index [%s] - mapping [%s]"`. If you want to omit index names, maybe something like `"Different mappings are not allowed for the same field: found [%s] and [%s]"`. Finally, please use `Locale.ROOT` with `String.format`.

Agree. Maybe we can make this easier, just give index name, field name/type and expected type. I think these should be sufficient for customer to understand what's happening.

@arsen-es btw, we may need a utility method if you think passing `Locale.ROOT` is required for each call to `String.format`.

Reply:
@arsen-es The message structure is taken from existing ElasticSearch SQL Exception

```
{
  "error": {
    "root_cause": [
      {
        "type": "verification_exception",
        "reason": "Found 1 problem(s)\nline 1:8: Cannot use field [field_1] due to ambiguities being mapped as [2] incompatible types: [text] in [test_1], [long] in [test_2]"
      }
    ],
    "type": "verification_exception",
    "reason": "Found 1 problem(s)\nline 1:8: Cannot use field [field_1] due to ambiguities being mapped as [2] incompatible types: [text] in [test_1], [long] in [test_2]"
  },
  "status": 400
}
```

Since we have already stripped out index names, we can't add index name to exception message.

Reply:
@dai-chen Since we can have multiple indices wit same field name but different mappings, We cannot identify a expected type. Any of them could be expected depending on users case.

Reply:
> @dai-chen Since we can have multiple indices wit same field name but different mappings, We cannot identify a expected type. Any of them could be expected depending on users case.

Sure. Sorry I meant the type in `fieldMappingValue` and `map.get(fieldName)` which we're comparing.

Reply:
@abbashus The current message says we are "mapping the ambiguities", which is incorrect. So let's fix it (I didn't mean you introduced the bad message).